gen/edge.ml

```diff
@@ -747,8 +748,9 @@ let fold_tedges f r =
   | Rf _|Fr _|Ws _|Irf _|Ifr _|Leave _|Back _| Hat -> true
   | _ -> false
 
-  let is_fetch e = match e.edge with
-  | Irf _|Ifr _ -> true
+  let is_fetch e = match e.edge,is_ifetch e.a1,is_ifetch e.a2 with
+  | Irf _,_,_|Ifr _,_,_
+  | Rf _,true,_| Fr _,_,true -> true
   | _ -> false
```

Comment:
However, if you add support for R and Ifetch, this will have to change.

Reply:
thanks for the suggestion! I have made the change. I don't think we need to change the logic with R however, as the W will always be to the same location, and it must have an I associated with it (because we can't have plain writes to a code location), this should still capture every case